### PR TITLE
Update to 18.04

### PR DESCRIPTION
**Pull request by ruebot**
**GitHub Issue**: https://github.com/Islandora-CLAW/CLAW/issues/848

# What does this Pull Request do?

- Update Vagrantfile to use 18.04 for Ubuntu builds
- Tweak package names, and required packages for 18.04
- Updates to get rid of some deprecation warnings
- Addresses https://github.com/Islandora-CLAW/CLAW/issues/848
- Requires https://github.com/Islandora-Devops/ansible-role-grok/pull/6

# How should this be tested?

- Requires https://github.com/Islandora-Devops/ansible-role-grok/pull/6
- `vagrant up`

# Interested parties
@Islandora-Devops/committers

### CODE REVIEW

**Comment by Natkeeran** [comment]
Deployed without issues.  Sanity tests with image works.  

@dannylamb want to do a final check?

**Comment by dannylamb** [comment]
`vagrant up`

**Comment by dannylamb** [comment]
failed on installing grok.  did any underlying roles change?

**Comment by Natkeeran** [comment]
@dannylamb 

Did you get the Islandora-Devops/ansible-role-grok#6 ?

**Comment by dannylamb** [comment]
Bingo.  Read the PR description Danny....

**Comment by dannylamb** [comment]
Ok, let's try this again :man_facepalming:

**Comment by dannylamb** [comment]
With https://github.com/Islandora-Devops/ansible-role-grok/pull/6 I'm getting

```bash
TASK [Islandora-Devops.grok : Install Grok dependencies (apt)] *****************
Monday 15 July 2019  16:51:50 -0300 (0:00:02.767)       0:35:07.447 *********** 
failed: [default] (item=[u'cmake', u'libpng16-dev', u'libtiff-dev', u'liblcms2-dev']) => {"ansible_loop_var": "item", "changed": false, "item": ["cmake", "libpng16-dev", "libtiff-dev", "liblcms2-dev"], "msg": "No package matching 'libpng16-dev' is available"}

```

**Comment by dannylamb** [comment]
FWIW I deleted the role and then updated requirements.yml to point to the branch for the PR.  Let me try again.

**Comment by dannylamb** [comment]
Yeah... turns out that when you provision a remote environment, it caches the roles in `roles/internal` as well.  So even if you blow up `roles/external` it resorts to using `roles/internal` before downloading a new version. :rage4: 

So that wold explain lots of things.  Sorry @ruebot.  Let's try this again.  `vagrant up`

**Comment by dannylamb** [comment]
Three derps later and I finally got it.  Provisions fine and passes my smoke tests.

I merged https://github.com/Islandora-Devops/ansible-role-grok/pull/6 and sliced a `2.0.0` release of the role.  @ruebot can you update the requirements.yml file to reference `2.0.0`?

**Comment by Natkeeran** [comment]
@dannylamb 

I've tested this and it works.  It does not look like we need the extra variable islandora_distro passed in as the only placed it was used has been removed.  Can you pls confirm?

**Comment by dannylamb** [comment]
Its used in the Vagrantfile to set up the linux user: https://github.com/Islandora-Devops/claw-playbook/blob/dev/Vagrantfile#L18

**Comment by Natkeeran** [comment]
We don't need it in the README as in the following commmand:
```
ansible-playbook -i inventory/production playbook.yml -e "islandora_distro=ubuntu/xenial64"
```

**Comment by dannylamb** [comment]
Yeah weird... Everything is really based off of `ansible_os_family`, which I can't seem to figure out how that gets defined by islandora_distro...  I'll have to dig.

**Comment by Natkeeran** [comment]
ansible_os_family seems like a built in variable.  

islandora_distro is only being used here: https://github.com/Islandora-Devops/claw-playbook/blob/65a7f9ec6b4217c6b4a5ed28925eaf15ff51b931/bootstrap.yml#L17, which is being removed with this PR.

**Comment by dannylamb** [comment]
I'll defer to @jonathangreen on https://github.com/Islandora-CLAW/CLAW/pull/1217.  I don't have enough experience with that sort of thing to speak authoritatively about it.

**Comment by dannylamb** [comment]
@Natkeeran Ok, so `ansible_os_family` is automagic.  I guess we were using ISLANDORA_DISTRO because it is not defined before ansible kicks off, and we need to do logic in the Vagrantfile with it.  If we're not using it outside of vagrant, then yes, delete it from the README example.

The only thing I want to make sure is that people can create a remote CENTOS environment without specifying that variable.  @seth-shaw-unlv?  Is it possible to build a remote CENTOS server without `ISLANDORA_DISTRO=centos/7`?

**Comment by Natkeeran** [comment]
@dannylamb 
We can merge this and find that out later as well via a separate ticket.

**Comment by seth-shaw-unlv** [comment]
It seems so @dannylamb. I recently ran just the tomcat.yml against a CentOS box and it set up correctly. I only use the variable when building a vagrant box.

**Comment by dannylamb** [comment]
@seth-shaw-unlv++ @Natkeeran++ Thanks guys.

@ruebot Can you chop off the `ISLANDORA_DISTRO` bits for provisioning a remote server?  Then let's do this :rocket:

**Comment by jonathangreen** [comment]
Islandora-CLAW/CLAW#1217 is merged now!
